### DISCUSSION TITLE
An easy-to-use subcommand completer for simple cases (especially for personalized scripts)

### DISCUSSION THREAD

**Discussion by julyfun** [fish-shell/fish-shell#10630] [Show and tell]
Hi fellows, I've implement a experimental completer in https://github.com/julyfun/jst.fish/blob/main/complete.fish that is easy to use: `__mfa.complete "the parent command"  "the subcommand to complete" "complete hints"` (There might be still some bugs)

Examples:

```
__mfa.complete "jst do" foo "Do something"
__mfa.complete "jst do" bar "Do some other thing"
```

Result (with a tab after `jst do `):

<img width="521" alt="image" src="https://github.com/user-attachments/assets/4408e9ed-4c27-4a52-b13f-bdd02530f9e9">

I've complete all my script functions with a single command:

```
# complete all functions that start with __jst with their `-d` option, and rename completions to jst
__mfa.complete-r __jst jst
```
<img width="469" alt="image" src="https://github.com/user-attachments/assets/8ec9e664-c626-4d03-8938-dcd526acbe3e">

You may try it and make suggestions and issues 😊

**Comment by ridiculousfish** [comment]
Thanks for sharing!
